pyccel/parser/syntactic.py

```diff
@@ -657,6 +659,13 @@ def _visit_FunctionDef(self, stmt):
 
                 i = i+1
 
+            if (len(arguments) != args_number):
+                msg = 'The number of arguments in the function ({}) {} and the types decorator ({}) {} don\'t match.'.format(len(arguments), arguments, args_number, decorators['types'].args)
+                if (len(arguments) < args_number):
+                    errors.report(msg, severity='warning')
+                else:
+                    errors.report(msg, severity='error')
```

Comment:
Severity should be fatal

Reply:
Ah, sorry I have just realised that we only use the information in the types decorator for the semantic step. Therefore it is not blocking in the syntactic step. The severity is therefore 'error' as you had before, not 'fatal'

Reply:
The difference between the 2 is that 'fatal' stops the current stage from completing. This is useful, for example, if we have not collected the `dtype` information and we will need it in this stage.
'error' allows the current stage to complete but doesn't move on to the next stage. So in this case that would allow all discoverable errors to be found in the syntactic stage but would not carry out the semantic stage where we will have problems due to a lack of information about the `dtype`

Reply:
thank you for the information now i understand the difference between the error and fatal severity, im gonna use error as severity.